python/cudf/cudf/core/index.py

```diff
@@ -175,6 +175,9 @@ def _apply_op(self, fn, other=None):
         else:
             return as_index(op())
 
+    def unique(self):
+        return as_index(self._values.unique())
```

Comment:
May want to special case RangeIndex to return self since it's always unique.

Reply:
Done